src/libData/AccountData/Account.cpp

```diff
@@ -126,36 +203,112 @@ bool Account::IncreaseNonce()
     return true;
 }
 
-void Account::SetStorage(string _k, string _mutable, string _type, string _v)
+void Account::SetStorageRoot(const h256& root)
+{
+    if (!isContract())
+        return;
+    m_storageRoot = root;
+    if (m_storageRoot != h256())
+    {
+        m_storage.setRoot(m_storageRoot);
+        m_prevRoot = m_storageRoot;
+    }
+}
+
+void Account::SetStorage(string _k, string _type, string _v, bool _mutable)
 {
-    RLPStream rlpStream(3);
-    rlpStream << _mutable << _type << _v;
-    m_storage.insert(
-        bytesConstRef(DataConversion::HexStrToUint8Vec(_k).data(), _k.size()),
-        rlpStream.out());
+    if (!isContract())
+        return;
+    RLPStream rlpStream(4);
+    rlpStream << _k << (_mutable ? "True" : "False") << _type << _v;
+
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(DataConversion::StringToCharArray(_k));
+    const vector<unsigned char>& k_hash = sha2.Finalize();
+
+    m_storage.insert(dev::h256(k_hash), rlpStream.out());
+
     m_storageRoot = m_storage.root();
 }
 
-vector<string> Account::GetKeys()
+vector<string> Account::GetStorage(string _k) const
+{
+    if (!isContract())
+        return {};
+
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(DataConversion::StringToCharArray(_k));
+    const vector<unsigned char>& k_hash = sha2.Finalize();
```

Comment:
Seems it's a duplicate code segment. We could reuse this via an internal function in your private namespace in  this file or escalate it into a global utility function